tools/src/test/resources/dev/run-qualification-tests.sh

```diff
@@ -0,0 +1,395 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2022, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+YELLOW='\e[33;1m'
+RED='\e[31;1m'
+GREEN='\e[32;1m'
+ENDCOLOR='\e[0m'
+
+MODULE_NAME="tools"
+RELATIVE_QUAL_LOG_PATH="src/test/resources/spark-events-qualification"
+RELATIVE_PROF_LOG_PATH="src/test/resources/spark-events-profiling"
+RELATIVE_QUAL_REF_PATH="src/test/resources/QualificationExpectations"
+RELATIVE_TOOLS_JAR_PATH="tools/target/spark311/"
+RELATIVE_COMMON_JAR_PATH="common/target/spark330/"
+
+
+PROJECT_ROOT=""
+MODULE_PATH=""
+PROF_LOG_DIR=""
+QUAL_LOG_DIR=""
+QUAL_REF_DIR=""
+JVM_DEFAULT_HEAP_SIZE="10g"
+
+# get the directory of the script
+WORK_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]:-$0}"; )" &> /dev/null && pwd 2> /dev/null; )";
+BATCH_OUT_DIR="${WORK_DIR}/qualification-output"
+CSV_OUT_DIR="${BATCH_OUT_DIR}/csv"
+RUNS_OUT_DIR="${BATCH_OUT_DIR}/runs-output"
+
+java_cp=""
+rapids_jar_home=""
+script_out_dir=""
+qual_events_dir=""
+prof_events_dir=""
+qual_expectations_dir=""
+
+heap_size=$JVM_DEFAULT_HEAP_SIZE
+declare -A qualification_path_map
+
+show_help()
+{
+   # Display Help
+   log_msg
+   log_msg "Run the Qualification tool to generate CSV files to update/replace Qualification expectations set."
+   log_msg "The output folder of the CSV files is test/resources/dev/qualification-output/csv"
+   log_msg
+   log_msg "The script compares the CSV files. If pairs of files do not match or the directories are"
```

Comment:
this sounds a bit weird, the whole point of the script is to update the results, so having it fail seems weird.  Perhaps we should say something like compares and reports on diffs. 
I assume user is responsible for looking at those and copying files if expected output changes

Reply:
I will rephrase the comment to clarify.
The script will run all the tests and logging the result of the comparison of each unit test.
at the end, if the two folders do not match (all unit tests match), it will log the diff result of the two folders, and `exit 1`.
I used `exit 1` here in case someone wants to call the script programmatically. So instead of grepping the log to check against the output, the caller can just check the return of the script.

Reply:
I updated the description of the script.